lib/controller.js

```diff
@@ -170,6 +177,134 @@ Tessel.get = function(opts) {
+
+    if (opts.timeout && typeof opts.timeout === 'number') {
+      // Stop the search after that duration
+      setTimeout(stopSeeker, opts.timeout * 1000);
+    } else {
+      // default to 3 seconds searching
+      setTimeout(stopSeeker, 3000);
+    }
+    logs.info('Searching accessable Tessels ...');
+
+    var seeker = new discover.TesselSeeker().start();
+    var tessels = [];
+
+    // When we find Tessels
+    seeker.on('tessel', function(tessel) {
+      controller.closeTesselConnections(tessel)
+        .then(function() {
+          var known = false;
+          if (tessels.length >= 1) {
+            for (var i in tessels) {
+              if (tessel.connections[0].ip === tessels[i].connections[0].ip && tessel.connections[0].connectionType === 'LAN') {
+                // Your network topology causes a single tessel is found twice
+                debug('known = true');
+                known = true;
+              }
+            }
+
+          }
+          if (known) {
+            debug('Due to your Network-Topology ' + tessel.name + ' is found twice! (' + tessel.connections[0].ip + ')');
+          } else {
+            // Add it to our array
+            tessels.push(tessel);
+          }
+        });
+    });
+
+    seeker.on('error', function(e) {
+      reject(e);
+    });
+
+    function stopSeeker() {
+      try {
+        // If there were no Tessels found
+        if (!tessels || tessels.length === 0) {
+          // Report the sadness
+          debug('We are searching the following networks:\n', networkInterfaces);
+          debug('\n(Important: Check your Netmask - we do not follow gateways for discovering!)');
+          logs.warn('No Tessels found (DEBUG=controller t2 root # might help!)');
+          resolve();
+        }
+        seeker.stop();
+        seeker = null; // preventing memory leaks
+        return resolve(tessels);
+      } catch (e) {
+        reject(e);
+      }
+    }
+  });
+};
+/*
+  The T2 root command is used to login into the firmware and gaining superuser access.
+  The security is provided by RSA - your Tessel get the keys while provisioning via USB.
+  
+  If you have only one Tessel, the T2 root command will login directly else there is 
+  a ncurses like menu you can select the Tessel you like to gain root access...
+
+  The structure of code and maybe unusual parts are paid due being testable.
+  The lightweight terminal-menu isn't written testable what needs a little bit hacking.
+
+  Finally some parts are causing from my personnal learning curve about writing unit tests using sinon!
+*/
+controller.root = function(opts) {
+  // ~ conversion to home because spawn isn't able to handle this right
+  if (opts.path && opts.path.substring(0, 1) === '~') {
+    var home = process.env.HOME;
+    opts.path = opts.path.replace('~', home);
+  }
+  var rtm;
+  if (!opts.menu) {
+    // TODO: Tessel cooperate identity conform menu required (color, ascii logo)
+    rtm = Menu({
```

Comment:
> If I understand it right, Tessel.list the Menu select event listener should set process.env.TESSEL to the selected tessel.name ?

A menu for selecting a Tessel should only be used with `Tessel.get`, not `Tessel.list`. Listing shows all available Tessels so there is no need to select one at that time. Let me know if that doesn't make sense. 

> An other idea would be a .tessel/defaultTessel.tmp with the tessel.name in it ?!

@Student007 I don't recommend storing state in a file across CLI calls because things can get hairy if you run two commands at once or another program is using the CLI API. In the latter case, you could have two processes writing to the file at the same time. 

When [logAndFinish](https://github.com/tessel/t2-cli/blob/master/lib/controller.js#L150-L169) get called after the search for a Tessel, if no `tessel` is passed in as the arg, it means multiple possible Tessels were found and this is where users should be able to select which they'd prefer. We should also print out a message that indicates that they can use the `--name` flag to avoid the manual selection process.